SparkPost/2web2ui
# TR-1374 - Templates Editor V2 Test Data

[TR-1374](https://jira.int.messagesystems.com/browse/TR-1374)

### What Changed
- Added new templates actions that update test data using `window.localStorage` instead of `localforage` (`createV2`, `updateV2`, `publishV2`, `sendPreviewV2`)
- Added new actions to update, retrieve, and delete test data from `localStorage` (`setTestDataV2`, `deleteTestDataV2`, `getTestDataV2`)
- Added new `EditTestDataSection` component to render the test data tab
- Updated the editor theme to `tomorrow_night_eighties` because it looks super sweet
- Updated background color of editor wrapper to match the new syntax theme
- Updated `PreviewSection` component to conditionally render the preview depending on what kind of data has been provided when the user has selected the `test_data` tab
- Adds new `useEditorTestData` hook that returns `testData`, `parsedTestData`, and the `setTestData` function
- Updated AMP linting script from Google
- Added defaults for template options in the event no options are received from the accounts endpoint

### How To Test
1. Navigation to `/templatesV2`
2. Check that `Test Data` tab renders on the lefthand side and renders the test data editor when active
3. Check that test data renders in the preview by adding this HTML:
```html
<h1>Hello {{world}}</h1>
```

and this test data:
```json
{
  "substitution_data": {
    "world": "hello"
  },
  "metadata": {},
  "options": {}
}
```

And the preview should render as:

> Hello hello

4. When the template is a "DRAFT", edit the test data, click "Save Draft" in the top right menu and verify that data is preserved when that particular template mounts and unmounts.
5. When in "DRAFT" click "Save and Publish" and submit the modal - verify that the new published version of the template has the same dest data as the draft version that produced it
6. Verify that deleting a template removes the test data from local storage by verifying in the console with `window.localStorage`
7. Verify that duplicating a template within the list page view also duplicates the test data when the copy is created
8. Verify that duplicating a template within the template view using the top right menu also duplicates the test data when the copy is created

### To Do
- [x] Add tests for updated actions
- [x] Get assistance testing `useEditorTestData`
